### PR TITLE
Containerize

### LINKED ISSUE #5: `devenv build-image`
Build containers based on `processes` specification and https://github.com/nlewo/nix2container

### PR DESCRIPTION
**Pull request by domenkozar**
Fixes #5

### CODE REVIEW

**Review by nlewo — COMMENTED** [review]
Maybe some people would ask to use a non root user by default since it's known as a good practice. If it is the case, this is supported by nix2container (i should add an example btw).

File: src/modules/builds/container.nix [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        then throw ''You need to set `name = "myproject";` to be able to generate a container.''
+        else config.name;
+      tag = cfg.version;
+      copyToRoot = [
+        (pkgs.runCommand "create-paths" { } ''
+          mkdir -p $out/tmp
+        '')
+        (pkgs.buildEnv {
+          name = "root";
+          paths = [
```
[the remark this answers is not in the document]
Comment (domenkozar) [reply src/modules/builds/container.nix:100]: There are a few optimizations like this that I'd like to do, but after this feature is shipped :)

[the remark this answers is not in the document]
Comment (nlewo) [reply src/modules/builds/container.nix:100]: Sure, i totally understand! (and that's why i said "FYI").

[the remark this answers is not in the document]
Comment (domenkozar) [reply src/modules/builds/container.nix:100]: Thanks for that though!

[the remark this answers is not in the document]
Comment (domenkozar) [reply src/modules/builds/container.nix]: https://github.com/cachix/devenv/issues/441

File: docs/containers.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+```nix title="devenv.nix"
+{
+  name = "simple-python-app";
+
+  languages.python.enable = true;
+}
+```
+
+Generate a container specification that enters the envirinoment:
```
[the remark this answers is not in the document]
Comment (nlewo) [reply docs/containers.md]: Maybe printing an help message saying the container spec has been built and what the user could do with it.

**Pull request merged**